From 5ec9f99f01f1b69db2254c9d1453c37c791d75a2 Mon Sep 17 00:00:00 2001
From: Kishore Kadiyala <kishore.kadiyala@intel.com>
Date: Mon, 29 Feb 2016 22:30:43 +0530
Subject: [PATCH 1/3] iio: accel: kxcjk-1013: Array of size 3 may use index
 value 3

Klockworks has reported an critical where an Array
'odr_start_up_times' of size 3 may use index value(s) 3.

Change-Id: I7811ac85ce15d16330a9878cc43eb92fa8fd64d2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-21888
Signed-off-by: Kishore Kadiyala <kishore.kadiyala@intel.com>
Reviewed-on: https://android.intel.com:443/475943
---
 drivers/iio/accel/kxcjk-1013.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/iio/accel/kxcjk-1013.c b/drivers/iio/accel/kxcjk-1013.c
index 2695dfd..50b9102 100644
--- a/drivers/iio/accel/kxcjk-1013.c
+++ b/drivers/iio/accel/kxcjk-1013.c
@@ -1231,6 +1231,9 @@ static int kxcjk1013_probe(struct i2c_client *client,
 	} else
 		return -ENODEV;
 
+	if (data->chipset >=  KX_MAX_CHIPS)
+		return -ENODEV;
+
 	ret = kxcjk1013_chip_init(data);
 	if (ret < 0)
 		return ret;
-- 
1.9.1

